File: Python/Lab_5/Lab_5.py
from lib import *

num_ind_goods = int(input("How many industrial products do you want create: "))
ind_goods = create_ind_goods(num_ind_goods)

num_eating_goods = int(input("How many eating products do you want create: "))
eating_goods = create_eating_goods(num_eating_goods)

time = datetime.now()
current_time = TDate(time.day, time.month, time.year)

full_price = 0
for i in range(num_eating_goods):
    if eating_goods[i].is_spoiled_good(current_time):
        full_price += eating_goods[i].get_sum()

print("The total price of all spoiled products is:", full_price)

full_price = 0
for i in range(num_ind_goods):
    if ind_goods[i].get_conditions() == "IN_STOCK":
        full_price += ind_goods[i].get_sum()

print("The total price of all products that store IN_STOCK:", full_price)

File: C++/Lab_5/Lab_5.cpp
#include "Goods.h"

int main()
{
	cout << "How many industrial products do you want create: ";
	int numInd; cin >> numInd;
	IndustrialGoods* indGoods = createInd(numInd);

	cout << "How many eating products do you want create: ";
	int numEat; cin >> numEat;
	EatingGoods* eatGoods = createEat(numEat);

	struct tm current_time;
	time_t t = time(0);
	localtime_s(&current_time, &t);
	TDate currentTime(current_time.tm_mday, current_time.tm_mon + 1, 1900 + current_time.tm_year);
	
	int full_price = 0;
	for (int i = 0; i < numEat; i++) {
		if (eatGoods[i].isSpoiledGood(currentTime)) {
			full_price += eatGoods[i].get_sum();
		}
	}
	cout << "The total price of all spoiled products is: " << full_price << endl;

	full_price = 0;
	for (int i = 0; i < numInd; i++) {
		if (indGoods[i].get_conditions() == IN_STOCK) {
			full_price += indGoods[i].get_sum();
		}
	}

	cout << "The total price of all products that store IN_STOCK: " << full_price << endl;

	return 0;
}